modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientExperimentalCodegen.java

```diff
@@ -1106,71 +1119,432 @@ protected void addAdditionPropertiesToCodeGenModel(CodegenModel codegenModel, Sc
+            example = (String) schema.getDefault();
         }
 
-        if ("String".equalsIgnoreCase(type) || "str".equalsIgnoreCase(type)) {
-            if (example == null) {
-                example = p.paramName + "_example";
+        if (StringUtils.isNotBlank(example) && !"null".equals(example)) {
+            if (ModelUtils.isStringSchema(schema)) {
+                example = "'" + example + "'";
+            }
+            return fullPrefix + example;
+        }
+
+        if (schema.getEnum() != null && !schema.getEnum().isEmpty()) {
+            // Enum case:
+            example = schema.getEnum().get(0).toString();
+            if (ModelUtils.isStringSchema(schema)) {
+                example = "'" + escapeText(example) + "'";
+            }
+            if (null == example)
+                LOGGER.warn("Empty enum. Cannot built an example!");
+
+            return fullPrefix + example;
+        } else if (null != schema.get$ref()) {
+            // $ref case:
+            Map<String, Schema> allDefinitions = ModelUtils.getSchemas(this.openAPI);
+            String ref = ModelUtils.getSimpleRef(schema.get$ref());
+            if (allDefinitions != null) {
+                Schema refSchema = allDefinitions.get(ref);
+                if (null == refSchema) {
+                    return fullPrefix + "None";
+                } else {
+                    String refTitle = refSchema.getTitle();
+                    if (StringUtils.isBlank(refTitle) || "null".equals(refTitle)) {
+                        refSchema.setTitle(ref);
+                    }
+                    if (StringUtils.isNotBlank(schema.getTitle()) && !"null".equals(schema.getTitle())) {
+                        included_schemas.add(schema.getTitle());
+                    }
+                    return toExampleValueRecursive(refSchema, included_schemas, indentationLevel, prefix, exampleLine);
+                }
+            } else {
+                LOGGER.warn("allDefinitions not defined in toExampleValue!\n");
+            }
+        }
+        if (ModelUtils.isDateSchema(schema)) {
+            example = "datetime.datetime.strptime('1975-12-30', '%Y-%m-%d').date()";
+            return fullPrefix + example;
+        } else if (ModelUtils.isDateTimeSchema(schema)) {
+            example = "datetime.datetime.strptime('2013-10-20 19:20:30.00', '%Y-%m-%d %H:%M:%S.%f')";
+            return fullPrefix + example;
+        } else if (ModelUtils.isBinarySchema(schema)) {
+            example = "open('/path/to/file', 'rb')";
+            return fullPrefix + example;
+        } else if (ModelUtils.isByteArraySchema(schema)) {
+            example = "'YQ=='";
+            return fullPrefix + example;
+        } else if (ModelUtils.isStringSchema(schema)) {
+            // a BigDecimal:
+            if ("Number".equalsIgnoreCase(schema.getFormat())) {return fullPrefix + "1";}
+            // TODO why is Number handled here?
```

Comment:
That's a special case we added a while ago for `type:string, format:number` to map the type into BigDecimal in Java as `format: double` or `format:float` of `type:number` have been mapped accordingly. PM me if you need more info.